contentcuration/contentcuration/api.py

```diff
@@ -4,15 +4,39 @@
 import logging
 import os
 import re
+import hashlib
+import shutil
 from functools import wraps
 from django.db.models import Q, Value
 from django.db.models.functions import Concat
-from django.core.exceptions import ObjectDoesNotExist
+from django.core.exceptions import ObjectDoesNotExist, SuspiciousOperation
 from django.http import HttpResponse
 from kolibri.content import models as KolibriContent
 from le_utils.constants import content_kinds
 import contentcuration.models as models
 
+def write_file_to_storage(fobj, check_valid = False):
+    # Check that hash is valid
+    checksum = hashlib.md5()
+    for chunk in iter(lambda: fobj.read(4096), b""):
+        checksum.update(chunk)
+    filename, ext = os.path.splitext(fobj._name)
+    hashed_filename = checksum.hexdigest()
+    full_filename = "{}{}".format(hashed_filename, ext)
+    fobj.seek(0)
+
+    if check_valid and checksum.hexdigest() != filename:
```

Comment:
Rather than re-computing the hexdigest, can you not reference `hashed_filename` here?